spec/models/sms/sms_decoder_spec.rb

```diff
@@ -24,6 +24,38 @@
     assert_decoding_fail(body: "lasjdalfksldjal", error: "invalid_form_code")
   end
 
+  context 'with SMS Authentication enabled' do
+    before(:all) { create_form(questions: %w(integer), authenticate_sms: true) }
+
+    it 'should work with the correct code provided' do
+      auth_code = user.sms_auth_code
+      assert_decoding(body: "#{auth_code} #{@form.code} 1.17", answers: [17])
+    end
+
+    it 'should raise an error with an incorrect code provided' do
+      auth_code = 'n000'
+      assert_decoding_fail(body: "#{auth_code} #{@form.code} 1.17", error: "user_not_found")
+    end
+
+    it 'should raise an error if no code is provided' do
+      assert_decoding_fail(body: "#{@form.code} 1.17", error: "user_not_found")
+    end
+
+    it 'should lock account after 3 failed attempts' do
+      create_list(:sms_message, 3, user: user, auth_failed: true, type: "incoming")
+      assert_decoding_fail(body: "#{@form.code} 1.17", error: "account_locked")
+    end
```

Comment:
Can we get one additional test using Timecop that verifies that 3 failed attempts spaced out by more than window/threshold don't trigger the account locking?